src/misc.test-d.ts

```diff
@@ -61,6 +66,34 @@ if (hasProperty(overlappingTypesExample, 'foo')) {
   expectAssignable<Record<'baz', unknown>>(overlappingTypesExample);
 }
 
+const exampleErrorWithCode = new Error('test');
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+exampleErrorWithCode.code = 999;
+
+// Establish that trying to check for a custom property on an error results in faulre
+expectError(exampleErrorWithCode.code);
+
+// Using custom Error property is allowed after checking with `hasProperty`
+if (hasProperty(exampleErrorWithCode, 'code')) {
+  expectType<unknown>(exampleErrorWithCode.code);
```

Comment:
These tests were all failing prior to this change

Reply:
Hmm... I'm wondering if a good change in the future would be to allow typing the property being checked. Something like:

``` typescript
if (hasProperty<number>(exampleErrorWithCode, 'code')) {
  expectType<number>(exampleErrorWithCode.code);
}
```

I guess we'll have to see how `hasProperty` ends up getting used.

Reply:
> Hmm... I'm wondering if a good change in the future would be to allow typing the property being checked. Something like:
> 
> ```ts
> if (hasProperty<number>(exampleErrorWithCode, 'code')) {
>   expectType<number>(exampleErrorWithCode.code);
> }
> ```
> 
> I guess we'll have to see how `hasProperty` ends up getting used.

The problem with that is that we don't actually check the type, it's just being cast. It could be more useful if we combine it with a Superstruct struct.

```ts
import { number } from 'superstruct';

if (hasProperty(exampleErrorWithCode, 'code', number())) {
  expectType<number>(exampleErrorWithCode.code);
}
```

Reply:
@Mrtenz Ah — fair!

Reply:
We can combine `hasProperty` with a `typeof` check or another superstruct-based assertion easily enough. 

I don't see us using this often in situations where we know something about the property already. If a property is known an optional, we could just use `if (object.optionalProperty)` to see if it exists. Establishing more information about the type of this unknown property would need some sort of runtime check as well.